File: code/kt/src/onlinejudge/leetcode/solved/P1.kt
package onlinejudge.leetcode.solved

class P1 {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = HashMap<Int, Int>(nums.size)
        nums.forEachIndexed { index, number ->
            val other = target - number
            map[other]?.let {
                return intArrayOf(it, index)
            }
            map[number] = index
        }
        return intArrayOf()
    }
}

File: code/rs/src/onlinejudge/leetcode/solved/p1.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        for (i, &num1) in nums.iter().enumerate() {
            for (j, &num2) in nums.iter().enumerate() {
                if i == j {
                    continue;
                }
                if target == num1 + num2 {
                    return vec![i as i32, j as i32];
                }
            }
        }
        return vec![];
    }
}
